workflow/taskset/task_set_manager.go

```diff
@@ -0,0 +1,166 @@
+package taskset
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeutil "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/argoproj/argo-workflows/v3/pkg/apis/workflow"
+	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+	"github.com/argoproj/argo-workflows/v3/pkg/client/clientset/versioned/typed/workflow/v1alpha1"
+	informer "github.com/argoproj/argo-workflows/v3/pkg/client/informers/externalversions/workflow/v1alpha1"
+	argowait "github.com/argoproj/argo-workflows/v3/util/wait"
+	"github.com/argoproj/argo-workflows/v3/workflow/metrics"
+	"github.com/argoproj/argo-workflows/v3/workflow/util"
+)
+
+type QueueWorkflowFunc func(string)
+
+const (
+	workflowTaskSetWorkers = 8
+)
+
+type WorkflowTaskSetManager struct {
+	wfTaskSetClient   v1alpha1.ArgoprojV1alpha1Interface
+	wfTaskSetInformer informer.WorkflowTaskSetInformer
+	queueWorkflowFunc QueueWorkflowFunc
+	wfTaskSetQueue    workqueue.RateLimitingInterface
+}
+
+func NewWorkflowTaskSetManager(client v1alpha1.ArgoprojV1alpha1Interface, informer informer.WorkflowTaskSetInformer, queueFunc QueueWorkflowFunc, metrics *metrics.Metrics) *WorkflowTaskSetManager {
+	return &WorkflowTaskSetManager{
+		wfTaskSetClient:   client,
+		wfTaskSetInformer: informer,
+		queueWorkflowFunc: queueFunc,
+		wfTaskSetQueue:    metrics.RateLimiterWithBusyWorkers(workqueue.DefaultControllerRateLimiter(), "wf_task_set_queue"),
+	}
+}
+
+func (wfts WorkflowTaskSetManager) CreateTaskSet(ctx context.Context, wf *wfv1.Workflow, nodeId string, tmpl wfv1.Template) error {
+	key := fmt.Sprintf("%s/%s", wf.Namespace, wf.Name)
+	log.WithField("workflow", wf.Name).WithField("namespace", wf.Namespace).WithField("TaskSet", key).Infof("Creating TaskSet")
+
+	obj, exists, err := wfts.wfTaskSetInformer.Informer().GetIndexer().GetByKey(key)
```

Comment:
I think you should not check the informer, instead, when creating, check for 409 Conflict error - you MUST check that error anyway - so just od that and don't bother with informer check

Reply:
I liked your suggestion but after conflict error code need to call  `obj, exists, err := wfts.wfTaskSetInformer.Informer().GetIndexer().GetByKey(key)` for update the object. I didn't see the advantage.

Reply:
I think your suggestion is good. I changed the implementations. I faced an issue when parallel HTTP templates in workflow.  Informer takes time to update the object.